src/main/java/io/cdap/plugin/gcp/gcs/source/GCSSource.java

```diff
@@ -169,18 +170,42 @@ public GCSSourceConfig() {
       this.copyHeader = false;
     }
 
+    @Override
+    public void validate() {
+      // no-op
+    }
+
     public void validate(FailureCollector collector) {
       super.validate(collector);
       // validate that path is valid
-      if (!containsMacro("path")) {
-        GCSPath.from(path);
+      if (!containsMacro(NAME_PATH)) {
+        try {
+          GCSPath.from(path);
+        } catch (IllegalArgumentException e) {
+          collector.addFailure(e.getMessage(), null).withConfigProperty(NAME_PATH).withStacktrace(e.getStackTrace());
+        }
+      }
+      if (!containsMacro(NAME_FILE_SYSTEM_PROPERTIES)) {
+        try {
+          getFileSystemProperties();
+        } catch (Exception e) {
+          collector.addFailure("File system properties must be a valid json.", null)
+            .withConfigProperty(NAME_FILE_SYSTEM_PROPERTIES).withStacktrace(e.getStackTrace());
+        }
+      }
+      if (!containsMacro(NAME_FILE_REGEX)) {
+        try {
+          getFilePattern();
+        } catch (IllegalArgumentException e) {
+          collector.addFailure("Provide a valid regular expression.", null)
```

Comment:
this won't say what is wrong with the pattern, so the user won't be able to adjust accordingly